pkgs/development/interpreters/lua-5/hooks/setup-hook.sh

```diff
@@ -27,7 +22,14 @@ addToLuaSearchPathWithCustomDelimiter() {
   # allowing relative modules to be used even when there are system modules.
   if [[ ! -v "${varName}" ]]; then export "${varName}=;;"; fi
 
+  # export only if the folder contains lua files
+  shopt -s globstar
+
+  for _file in ${absPattern/\?/\*\*}; do
   export "${varName}=${!varName:+${!varName};}${absPattern}"
+    return;
```

Comment:
i'm not super familiar with how `shopt` works, but if we return here, won't `shopt` stay set?

also, the indentation on this line doesn't quite match.

Reply:
indeed ty. so other hooks dont seem to care about disabling it but I remember reading something about not enabling it by default so I tried to disable it. I 've disabled it before the return and fixed the indentation.

With this PR, the setup-hook is still broken as it doesn't respect the interpreter LUA_PATH but what we could do to finally fix it is to remove the nix LuaPathSearchPaths and generate it from the relative paths in the default LUA_PATH. i.e., in the hook replace:
`  for pattern in @luapathsearchpaths@; do`
with 
```
IFS=';'
patterns=()
read -a patterns <<< $(lua -e print(package.path))
# filter out absolute paths from patterns ?
for pattern in patterns; do
``` 
we could even save the resulting patterns into the lua derivation as nix artefact to speed up thing.

Reply:
> With this PR, the setup-hook is still broken as it doesn't respect the interpreter LUA_PATH

i'm pretty sure adding `;;` to LUA_PATH fixes that?  are you doing your testing on staging?